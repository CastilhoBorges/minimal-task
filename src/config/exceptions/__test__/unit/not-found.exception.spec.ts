import { HttpStatus } from '@nestjs/common';
import { NotFoundException } from '../../not-found.exception';

describe('NotFoundException', () => {
  it('should create an instance of NotFoundException with correct properties', () => {
    const message = 'The requested resource could not be found';
    const metadata = { code: 'RESOURCE_NOT_FOUND' };

    const exception = new NotFoundException(message, metadata);

    expect(exception).toBeInstanceOf(NotFoundException);
    expect(exception.getResponse()).toEqual({
      name: 'NotFoundException',
      message,
      statusCode: HttpStatus.NOT_FOUND,
      metadata,
      stack: undefined,
    });
    expect(exception.getStatus()).toBe(HttpStatus.NOT_FOUND);
    expect(exception.getResponse()).toMatchSnapshot();
  });

  it('should create an instance with only message', () => {
    const message = 'The requested resource could not be found';

    const exception = new NotFoundException(message);

    expect(exception).toBeInstanceOf(NotFoundException);
    expect(exception.getResponse()).toEqual({
      name: 'NotFoundException',
      message,
      statusCode: HttpStatus.NOT_FOUND,
      metadata: undefined,
      stack: undefined,
    });
    expect(exception.getStatus()).toBe(HttpStatus.NOT_FOUND);
    expect(exception.getResponse()).toMatchSnapshot();
  });
});
