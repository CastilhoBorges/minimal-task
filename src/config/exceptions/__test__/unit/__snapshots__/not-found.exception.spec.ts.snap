// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundException should create an instance of NotFoundException with correct properties 1`] = `
{
  "message": "The requested resource could not be found",
  "metadata": {
    "code": "RESOURCE_NOT_FOUND",
  },
  "name": "NotFoundException",
  "stack": undefined,
  "statusCode": 404,
}
`;

exports[`NotFoundException should create an instance with only message 1`] = `
{
  "message": "The requested resource could not be found",
  "metadata": undefined,
  "name": "NotFoundException",
  "stack": undefined,
  "statusCode": 404,
}
`;
